src/pages/patientView/clinicalInformation/SampleGeneCache.spec.ts

```diff
@@ -0,0 +1,178 @@
+import {default as SampleGeneCache, SampleToEntrezListOrNull, Cache} from './SampleGeneCache';
+import { assert, default as chai } from 'chai';
+import sinon from 'sinon';
+
+chai.config.truncateThreshold = 0;
+
+type Data = {
+    sampleId: string;
+    entrezGeneId: number;
+    data: string;
+};
+
+describe('SampleGeneCache', () => {
+
+    let data:Data[];
+
+    let sampleIds:string[];;
+
+    let fetchPromise:Promise<Data[]>;
+
+    let initialCache:Cache<Data> = {};
+
+    class CallbackCache extends SampleGeneCache<Data> {
+        public async populate(sampleToEntrezList:SampleToEntrezListOrNull) {
+            return super.populate(sampleToEntrezList);
+        }
+        protected fetch(sampleToEntrezList:SampleToEntrezListOrNull, callback:(...args:any[])=>void) {
+            callback(sampleToEntrezList);
+            return fetchPromise;
+        }
+    }
+
+    class NumberCallbackCache extends SampleGeneCache<Data> {
+        public async populate(sampleToEntrezList:SampleToEntrezListOrNull) {
+            return super.populate(sampleToEntrezList);
+        }
+        protected fetch(sampleToEntrezList:SampleToEntrezListOrNull, n:number, callback:(x:number)=>void) {
+            callback(n);
+            return Promise.resolve([]);
+        }
+    }
+
+    class FailingCache extends SampleGeneCache<Data> {
+        public async populate(sampleToEntrezList:SampleToEntrezListOrNull) {
+            return super.populate(sampleToEntrezList);
+        }
+        protected fetch(sampleToEntrezList:SampleToEntrezListOrNull) {
+            return Promise.reject("FAIL");
+        }
+    }
+
+    before(()=>{
+        data = [{sampleId:"sample1", entrezGeneId:0, data:"a"}, {sampleId:"sample1", entrezGeneId:1, data:"b"},
+                {sampleId:"sample2", entrezGeneId:0, data:"c"},{sampleId:"sample2", entrezGeneId:2, data:"d"}];
+
+        sampleIds = Object.keys(data.reduce((map:any, d:Data)=>{ map[d.sampleId] = true; return map; }, {}));
```

Comment:
Sorry, just realized you want `sampleIds = _.union(data.map(d => d.sampleId)` :)